museek/plugin/sanity_check_observation_plugin.py

```diff
@@ -220,3 +231,61 @@ def create_plots_of_scan_data(self, data: TimeOrderedData):
         plt.hist(data.elevation.squeeze.flatten(), bins=200)
         plt.xlabel('elevation')
         self.savefig(description='Elevation histogram of all dishes during scan.')
+
+
+    def check_closeness_to_sunrise_sunset(self, data: TimeOrderedData, report_writer: ReportWriter, observation_date):
+
+        """
+        Check the time difference between sunset/sunrise and start/end time.
+        :param data: the `TimeOrderedData` to check
+        :param report_writer: the `ReportWriter` object to handle the report
+        :param observation_date: the 'datetime' object referring to the observation date 
+
+        In a few cases, previous_setting() or next_rising() would give you the last or next day's 
+        sunset or sunrise date, so considering that sunset/rise times change by about half minute every day,
+        only hour, minute, and second information is used for calculating time difference       
+        """
+
+        observer = ephem.Observer()
+        observer.lat = data.antennas[0].ref_observer.lat
+        observer.lon = data.antennas[0].ref_observer.long
+
+        # Set the observer's date in UTC
+        observer.date = observation_date - timedelta(hours=2)
+
+        # Calculate sunset and sunrise times
+        sunset_time = observer.previous_setting(ephem.Sun())  # Sunset
+        sunrise_time = observer.next_rising(ephem.Sun())  # Sunrise
+        
+        # Convert times to South Africa local timezone
+        sunset_local_time = datetime.strptime(str(sunset_time), "%Y/%m/%d %H:%M:%S") + timedelta(hours=2)
+        sunrise_local_time = datetime.strptime(str(sunrise_time), "%Y/%m/%d %H:%M:%S") + timedelta(hours=2)
+
+        start_local_time = datetime.utcfromtimestamp(float(data.original_timestamps[0])) + timedelta(hours=2)
+        end_local_time = datetime.utcfromtimestamp(float(data.original_timestamps[-1])) + timedelta(hours=2)
+
+        time_difference_sunset = abs(start_local_time - sunset_local_time).total_seconds()%SECONDS_IN_ONE_DAY/60.
+        time_difference_sunrise = abs(end_local_time - sunrise_local_time).total_seconds()%SECONDS_IN_ONE_DAY/60. 
```

Comment:
if the sun rises 31 minutes before the observation ends, will this check work?

Reply:
It seems to me that the check only works correctly if the sun rises no later than 29 minutes before observation end, or if it sets no more than 29 minutes after observation start. Otherwise I think the check will pass even though it should not.

The best way would be to move this functionality into the `util` folder and to write unittests

Reply:
I reverted these two lines to an earlier version because, upon conducting tests, I discovered that using %SECONDS_IN_ONE_DAY could lead to issues when the time difference is less than 86400 seconds. The older version functions without any problems.

Reply:
It still uses abs(time difference). So if I set, for example, 5 minutes as a threshold and the observation starts 6 minutes before sunrise, it seems to me the test passes, no?

Reply:
method is updated